lib/AST/ModuleLoader.cpp

```diff
@@ -54,4 +59,73 @@ DependencyTracker::getClangCollector() {
   return clangCollector;
 }
 
+static bool findOverlayFilesInDirectory(SourceLoc diagLoc, StringRef path,
+                                        ModuleDecl *module,
+                                        DependencyTracker * const tracker) {
+  using namespace llvm::sys;
+  using namespace file_types;
+
+  ASTContext &ctx = module->getASTContext();
+  auto fs = ctx.SourceMgr.getFileSystem();
+
+  std::error_code error;
+  for (auto dir = fs->dir_begin(path, error);
+       !error && dir != llvm::vfs::directory_iterator();
+       dir.increment(error)) {
+    StringRef file = dir->path();
+    if (lookupTypeForExtension(path::extension(file)) != TY_SwiftOverlayFile)
+      continue;
+
+    module->addCrossImportOverlayFile(file);
+
+    // FIXME: Better to add it only if we load it.
```

Comment:
Not asking you to do this right now, but any thoughts on how we might propagate sufficient information to track this in a fine-grained manner?

Reply:
We'd need to thread `tracker` through to `OverlayFile` and have it only add the dependency when `loadOverlayModuleNames()` is called. Nothing difficult, just annoying.